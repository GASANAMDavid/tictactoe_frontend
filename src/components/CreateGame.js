import React from "react";
import Select from "react-select";
import Button from "./Button";
import { Form } from "react-bootstrap";

const languageOptions = [
  { value: "en", label: "English" },
  { value: "fr", label: "French" },
];

const modeOptions = [
  { value: 1, label: "Human V Random Computer" },
  { value: 2, label: "Human V Intelligent Computer" },
];

const boardSizeOptions = [
  { value: 3, label: 3 },
  { value: 4, label: 4 },
  { value: 5, label: 5 },
];

const CreateGame = ({ onSelectLanguage, onCreateGame }) => {
  const [isLanguageSet, setIsLanguageSet] = React.useState(false);
  const [language, setLanguage] = React.useState("");
  const [playerName, setPlayerName] = React.useState("");
  const [symbol, setSymbol] = React.useState("");
  const [gameMode, setGameMode] = React.useState(1);
  const [boardSize, setBoardSize] = React.useState();

  const handleChange = (lang) => {
    setIsLanguageSet(true);
    onSelectLanguage(lang);
    setLanguage(lang);
    
  };

  const handleClick = () => {
    console.log(symbol);
    console.log(language);
    console.log(playerName);
    console.log(gameMode);
    console.log(boardSize);
    onCreateGame()
  };

  return (
    <div className='create-game'>
      {!isLanguageSet ? (
        <label>
          choose language
          <Select
            className='select-language'
            options={languageOptions}
            value={language}
            onChange={(event) => handleChange(event.value)}
          />
        </label>
      ) : (
        <Form>
          <Form.Row>
            <label>
              Player name:
              <input
                className='input-player-name'
                placeholder='Enter name'
                value={playerName}
                onChange={(event) => {
                  setPlayerName(event.target.value);
                }}
              />
            </label>
          </Form.Row>
          <Form.Row>
            <label>
              Symbol:
              <input
                className='input-symbol'
                value={symbol}
                placeholder='Enter player symbol'
                onChange={(event) => setSymbol(event.target.value)}
              />
            </label>
          </Form.Row>
          <Form.Row>
            <label>
              Game mode:
              <Select
                className='select-game-mode'
                options={modeOptions}
                onChange={(event) => setGameMode(event.value)}
                value={gameMode}
              />
            </label>
          </Form.Row>
          <Form.Row>
            <label>
              BoardSize:
              <Select
                className='select-board-size'
                options={boardSizeOptions}
                isClearable
                onChange={(event) => setBoardSize(event.value)}
                value={boardSize}
              />
            </label>
          </Form.Row>
          <Button color='#2699ab' text='Create game' onClick={handleClick} />
        </Form>
      )}
    </div>
  );
};

export default CreateGame;
