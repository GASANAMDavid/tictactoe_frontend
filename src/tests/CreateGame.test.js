// import React from "react";
import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import CreateGame from "../components/CreateGame";

// jest.mock("react", () => ({
//   ...jest.requireActual("react"),
//   useState: jest.fn(),
// }));

describe("CreateGame", () => {
  let createGame;
  const mockGetSelectedLanguageContent = jest.fn();
  const lang = "en";
  beforeEach(() => {
    createGame = shallow(
      <CreateGame onSelectLanguage={mockGetSelectedLanguageContent} />
    );
    createGame.find(".select-language").simulate("change", { value: lang });
  });

  it("renders correctly", () => {
    expect(toJson(createGame)).toMatchSnapshot();
  });

  describe("when language is not set", () => {
    it("calls onSelectLanguage with the selected_language", () => {
      expect(mockGetSelectedLanguageContent).toHaveBeenCalledWith(lang);
    });
  });

  describe("when language is set", () => {
    it("should have input for player name", () => {
      // const name = "Manzi";
      // const mockSetPlayerName = jest.fn();
      // React.useState = () => ["", mockSetPlayerName];
      // createGame
      //   .find(".input-player-name")
      //   .simulate("change", { target: { value: name } });

      // expect(mockSetPlayerName).toHaveBeenCalledWith(name); //
      expect(createGame.find(".input-player-name").exists()).toBe(true);
    });

    it("should have input for symbol", () => {
      expect(createGame.find(".input-symbol").exists()).toBe(true);
    });

    it("should allow user to choose game mode", () => {
      expect(createGame.find(".select-game-mode").exists()).toBe(true);
    });

    it('allows user to choose boardSize', () => {
      expect(createGame.find('.select-board-size').exists()).toBe(true)
    })

    describe('submit button', () => {
      it('has a button to create game', () => {
        expect(createGame.find('Button').exists()).toBe(true)
      })
      
    })
    
  });
});
