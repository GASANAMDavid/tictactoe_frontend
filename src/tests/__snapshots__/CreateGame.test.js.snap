// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateGame renders correctly 1`] = `
<div
  className="create-game"
>
  <Form
    inline={false}
  >
    <FormRow>
      <label>
        Player name:
        <input
          className="input-player-name"
          onChange={[Function]}
          placeholder="Enter name"
          value=""
        />
      </label>
    </FormRow>
    <FormRow>
      <label>
        Symbol:
        <input
          className="input-symbol"
          onChange={[Function]}
          placeholder="Enter player symbol"
          value=""
        />
      </label>
    </FormRow>
    <FormRow>
      <label>
        Game mode:
        <StateManager
          className="select-game-mode"
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "Human V Random Computer",
                "value": 1,
              },
              Object {
                "label": "Human V Intelligent Computer",
                "value": 2,
              },
            ]
          }
          value={1}
        />
      </label>
    </FormRow>
    <FormRow>
      <label>
        BoardSize:
        <StateManager
          className="select-board-size"
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          isClearable={true}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": 3,
                "value": 3,
              },
              Object {
                "label": 4,
                "value": 4,
              },
              Object {
                "label": 5,
                "value": 5,
              },
            ]
          }
        />
      </label>
    </FormRow>
    <Button
      color="#2699ab"
      onClick={[Function]}
      text="Create game"
    />
  </Form>
</div>
`;
